sympy/ntheory/ecm.py

```diff
@@ -182,36 +192,56 @@ def _ecm_one_factor(n, B1=10000, B2=100000, max_curve=200, seed=None):
     ==========
 
     n : Number to be Factored
-    B1 : Stage 1 Bound
-    B2 : Stage 2 Bound
+    B1 : Stage 1 Bound. Must be an even number.
+    B2 : Stage 2 Bound. Must be an even number.
     max_curve : Maximum number of curves generated
 
+    Returns
+    =======
+
+    integer : A non-trivial divisor of ``n``. If ``n`` is prime, then ``n``.
+
+    Raises
+    ======
+
+    ValueError
+        When we cannot find a non-trivial divisor of ``n``.
+
     References
     ==========
 
-    .. [1]  Carl Pomerance and Richard Crandall "Prime Numbers:
-        A Computational Perspective" (2nd Ed.), page 344
+    .. [1] Carl Pomerance, Richard Crandall, Prime Numbers: A Computational Perspective,
+           2nd Edition (2005), page 344, ISBN:978-0387252827
     """
-    n = as_int(n)
-    if B1 % 2 != 0 or B2 % 2 != 0:
-        raise ValueError("The Bounds should be an even integer")
-    sieve.extend(B2)
-
     randint = _randint(seed)
-
     if isprime(n):
         return n
 
-    D = sqrt(B2)
-    beta = [0]*(D + 1)
-    S = [0]*(D + 1)
+    # When calculating T, if (B1 - 2*D) is negative, it cannot be calculated.
```

Comment:
It's not clear to me how this criteria is being used in the calculation of D (if that's where it is being used).

Reply:
For `D`, we would normally like to adopt `sqrt(B2)`. However, `mont_ladder` is not supposed to be given a negative number as its argument. Hence, `B1 - 2*D` must be positive. That is, `B1//2 > D` is required.

Therefore, we put an upper bound on `D`.